<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

  
<!--   <include file="$(find openni2_launch)/launch/openni2.launch"/> -->
<!--   <arg name="robotID" default="44" /> -->
  <arg name="robotID" default=""/>
  <arg name="mateID" default=""/>

  
  <arg name="selfCamInfo" default="/camera$(arg robotID)/rgb/camera_info" />  
  <arg name="mateCamInfo" default="/camera$(arg mateID)/rgb/camera_info" /> 

     
  <node name="tgt_stereo_detector$(arg robotID)" pkg="tgt_stereo_detection" type="tgt_stereo_detection" output="screen">
    <!-- This area is in pixels -->    
    <param name="selfCamInfoTopic" value="$(arg selfCamInfo)" type="string" />
    <param name="mateCamInfoTopic" value="$(arg mateCamInfo)" type="string" />
    <param name="selfRobotPoseTopic" value="/TeleKyb/Vicon/UAV_Octo$(arg robotID)/UAV_Octo$(arg robotID)" type="string" />
    <param name="mateRobotPoseTopic" value="/TeleKyb/Vicon/UAV_Octo$(arg mateID)/UAV_Octo$(arg mateID)" type="string" />
    <param name="selfCamBaseLink" value="camera$(arg robotID)_link" type="string" /> 
    <param name="selfCamRGBFrameLink" value="camera$(arg robotID)_rgb_frame" type="string" />
    <param name="selfCamRGBOpticalFrameLink" value="camera$(arg robotID)_rgb_optical_frame" type="string" />
    <param name="selfRobotBaseLink" value="base_link_robot$(arg robotID)" type="string" />
    <param name="mateCamBaseLink" value="camera$(arg mateID)_link" type="string" /> 
    <param name="mateCamRGBFrameLink" value="camera$(arg mateID)_rgb_frame" type="string" />
    <param name="mateCamRGBOpticalFrameLink" value="camera$(arg mateID)_rgb_optical_frame" type="string" />    
    <param name="mateRobotBaseLink" value="base_link_robot$(arg mateID)" type="string" />    
    <param name="P1_cam1" value="camera$(arg robotID)_P1_cam1" type="string" /> 
    <param name="P2_cam1" value="camera$(arg robotID)_P2_cam1" type="string" />
    <param name="Q1_cam2" value="camera$(arg robotID)_Q1_cam2" type="string" />    
    <param name="Q1_cam2_naive" value="camera$(arg robotID)_Q1_cam2_naive" type="string" />
    <param name="detectedObjectLink" value="detected_object_link_by_robot$(arg robotID)" type="string" /> 
    <param name="projectedObjectLink" value="projected_detected_object_link_by_robot$(arg robotID)" type="string" />  
    <param name="detectedObject_Wframe_Topic" value="detectedObject_Wframe_byRobot$(arg robotID)" type="string" /> 
    <param name="detectedObject_WframeNaive_Topic" value="detectedObject_WframeNaive_byRobot$(arg robotID)" type="string" />
    
    <param name="selfBlobTopicName" value="/tgt_detector$(arg robotID)/targetBlobInCamera$(arg robotID)" type="string" />
    <param name="mateBlobTopicName" value="/tgt_detector$(arg mateID)/targetBlobInCamera$(arg mateID)" type="string" />
   </node>

</launch>